### PR TITLE
Feature/EUI 3738 - add generic filter service and component

### PR DESCRIPTION
**Pull request by leepavlouhmcts**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

https://tools.hmcts.net/jira/browse/EUI-3738
https://tools.hmcts.net/jira/browse/EUI-3739

### CODE REVIEW

File: projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    this.filterService.persist(this.settings, this.config.persistence);
+  }
+
+  public cancelFilter() {
+    this.getSettings();
+  }
+
+  public isSelected(id: string, key: string): boolean {
+    if (this.settings) {
+      return this.settings.fields.some(x=> x.name == id) && this.settings.fields.some(x => x.value.some(x => x == key));
```
Comment (udaydenduluri33) [inline projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts:34]: I am not sure about this logic. using some inside some.

Comment (leepavlouhmcts) [reply projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts:34]: its just determining if the settings exist for that key. what would you prefer i use, indexof?

Comment (udaydenduluri33) [reply projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts:34]: My question was does it work. Do you have a test for it?

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Uday's comment just needs looking at

**Review by connorpgpmcelroy — APPROVED** [review]
Comment has been addressed by test. LGTM

**Review by udaydenduluri33 — APPROVED** [review]
LGTM
Agreed to fix the comments in tech debt JIRA later

**Pull request merged**